queries.sql

```diff
@@ -7,3 +7,53 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * from animals WHERE neutered=TRUE;
 SELECT * from animals WHERE name <> 'Gabumon';
 SELECT * from animals WHERE weight_kg >=10.4 and weight_kg <= 17.3;
+BEGIN;
+update animals set species='unspecified' where species is null;
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species='pokemon' WHERE species is NULL;
+COMMIT;
+
+BEGIN;
+DELETE * FROM animals
+ROLLBACK;
+
+
+BEGIN;
+-- Delete all animals born after Jan 1st, 2022.
+DELETE FROM animals WHERE date_of_birth > '2022/01/01';
+
+-- Create a savepoint for the transaction.
+SAVEPOINT sp1;
+
+-- Update all animals' weight to be their weight multiplied by -1.
+UPDATE animals SET weight_kg = weight_kg*-1;
+
+-- Rollback to the savepoint
+ROLLBACK TO sp1;
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+UPDATE animals SET weight_kg=(SELECT weight_kg FROM animals WHERE weight_kg < 0)*-1;
+
+-- Commit transaction
+COMMIT;
+
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) from animals WHERE escape_attempts=0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT name FROM animals WHERE escape_attempts=(SELECT MAX(escape_attempts) FROM animals);
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MIN(weight_kg), MAX(weight_kg) FROM animals;
```

Comment:
- [ ] Since we are trying to query the database for minimum and maximum weight of each type of animal, then it will be better if this is grouped by the species. Please kindly group by species.